### PR TITLE
Bump to 1.4.6

### PR DESCRIPTION
**Pull request by tupui**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by tupui** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,30 +1,28 @@
...
-    - numpy >=1.16
-    - pandas >=1.0.3
-    - scipy
-    - matplotlib-base
-    - pathos
+    - python >=3.8
+    - numpy >=1.20.3
+    - pandas >=1.1.2
+    - scipy >=1.7.3
+    - matplotlib >=3.2.2
```
Comment (tupui) [inline recipe/meta.yaml]:
```suggestion
    - matplotlib-base
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,30 +1,28 @@
...
-  script: {{ python }} -m pip install . --no-deps -vv
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
 requirements:
   host:
     - pip
-    - python >=3.6
-    - setuptools >=38.3
-    - pyscaffold >=3.2.3
+    - python >=3.8
+    - hatchling>=1.8.1
```
Comment (tupui) [inline recipe/meta.yaml]:
```suggestion
    - hatchling >=1.8.1
```

**Comment by tupui** [comment]
(Closed reopened to trigger the CI again.)

**Review by ConnectedSystems — APPROVED** [review]
Thanks!

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,30 +1,28 @@
-{% set version = "1.4.5" %}
-{% set python = "python3" %}
+{% set version = "1.4.6" %}
 
 package:
   name: salib
   version: {{ version }}
 source:
   fn: SALib-{{ version }}.tar.gz
   url: https://pypi.io/packages/source/S/SALib/SALib-{{ version }}.tar.gz
```
Comment (tupui) [inline recipe/meta.yaml]:
```suggestion
  url: https://pypi.io/packages/source/s/salib/salib-{{ version }}.tar.gz
```

**Comment by tupui** [comment]
Great thanks @ConnectedSystems! It's already up.

I am surprised the platform thing was not updated. Must be because there are still refs to old versions for non no-arch. Not an issue.

**Pull request merged**